File: src/main/java/p00/Problem_0024.java
package p00;

import java.util.ArrayList;
import java.util.Arrays;

public class Problem_0024 {

	public static String solve() {

		int[] numbers = new int[10];
		int target = 999999;

		Integer[] array = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
		ArrayList<Integer> list = new ArrayList<>(Arrays.asList(array));
		StringBuilder solution = new StringBuilder();

		for (int i = 0; i < 10; i++) {

			int fac = faculty(9 - i);

			numbers[i] = target / fac;
			target = target % fac;

			solution.append(list.get(numbers[i]));
			list.remove(numbers[i]);
		}
		return solution.toString();
	}

	private static int faculty(int n) {
		int result = 1;

		for (int i = 2; i <= n; i++)
			result *= i;

		return result;
	}

}

File: src/main/kotlin/p00/Problem0024.kt
package p00

import java.util.*

class Problem0024 {
    fun solve(): String {
        val numbers = IntArray(10)
        var target = 999999
        val array = arrayOf(0, 1, 2, 3, 4, 5, 6, 7, 8, 9)
        val list = ArrayList(listOf(*array))
        val solution = StringBuilder()
        for (i in 0..9) {
            val fac = faculty(9 - i)
            numbers[i] = target / fac
            target %= fac
            solution.append(list[numbers[i]])
            list.removeAt(numbers[i])
        }
        return solution.toString()
    }

    private fun faculty(n: Int): Int {
        var result = 1
        for (i in 2..n) result *= i
        return result
    }
}